.github/workflows/publish-to-pypi.yml

```
@@ -0,0 +1,44 @@
+name: Publish Modin wheel to PyPI
+
+on:
+  schedule:
+    - cron: "42 0 * * WED"
+  push:
+    tags:        
+      - '*'
+  workflow_dispatch:
+
+jobs:
+  build-n-publish:
+    name: Build and publish Modin wheel to PyPI
+    environment: release
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write  # IMPORTANT: this permission is mandatory for trusted publishing
+
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+        fetch-tags: true
+    - name: Checkout latest git tag
+      run: git checkout $(git describe --tags "$(git rev-list --tags --max-count=1)")
+      if: github.event_name == 'push'
```

In my suggestion I swapped the order of these 😢 
```suggestion
      if: github.event_name == 'push'
      run: git checkout $(git describe --tags "$(git rev-list --tags --max-count=1)")
```

---

> In my suggestion I swapped the order of these 😢

Actually, I've adjusted your suggestion a bit to make the style similar to what is used in `ci.yml` :) Does it make sense?

---

Oh, is this valid too? If so I'm okay to merge this

---

> Oh, is this valid too? 

yes